### PR TITLE
ublaboo -> contributte (#1067)

### PR DESCRIPTION
**Pull request by lilianalillyy**
Renames namespace from `Ublaboo\DataGrid` to `Contributte\Datagrid`  (issue #1067) - package name not yet changed. 

Script that generates compatibility.php: https://gist.github.com/lilianalillyy/ee8de66b416fd1d680c58eba3f641d15

### CODE REVIEW

**Comment by radimvaculik** [comment]
@lilianalillyy Thanks for PR! 🎉 Could you make CI green, please?

**Comment by lilianalillyy** [comment]
This should be everything, the only things that have been left intentionally

1. `ublaboo/datagrid` library name (Composer) and `ublaboo-datagrid` (NPM) - This I'm leaving to @f3l1x / @paveljanda, who have to publish new packages (imo, the current packages should be deprecated though not removed completely, with a deprecation notice to use `contributte/datagrid`/`@contributte/datagrid` instead (both managers support this, even with the ability of referencing which package should be used as replacement). (Affected files: .docs/index,filters,assets.md, .docs/assets.md, README.md, package.json, composer.json)
2. CSS/JS assets - those are both subject to change already (rewrite)

As for translation keys, they are now prefixed with `contributte_`, rest stays the same. Before, they used the `ublaboo_` prefix, which will still function and will be automatically transformed to the contributte_ equivalent (a E_USER_DEPRECATED error will be triggered for the ublaboo_ keys). Implementation of that can be seen in SimpleTranslator.php.

**Comment by radimvaculik** [comment]
@lilianalillyy Could you make CI green, please?

**Comment by lilianalillyy** [comment]
`PHP Fatal error:  Declaration of Contributte\Datagrid\Filter\SubmitButton::getControl(object|string|null $caption = null): Nette\Utils\Html must be compatible with Nette\Forms\Controls\Button::getControl($caption = null): Nette\Utils\Html in /home/runner/work/datagrid/datagrid/src/Filter/SubmitButton.php on line 34`

It doesn't make sense to me, for reference:
```php
<?php
// ...
use Nette\Utils\Html;

// ...

public function getControl(string|object|null $caption = null): Html
{
 // ...
}
```

**Comment by radimvaculik** [comment]
Does it fail on your local, when `make cs` ?

**Comment by lilianalillyy** [comment]
nope, I only had an issue with the alphabetical sorting (which I've now fixed), this error didn't occur

**Comment by radimvaculik** [comment]
I just clone your repo, run `make` and `make cs` and got the error. Running on PHP 8.0 locally.

**Comment by lilianalillyy** [comment]
Removing vendor and reinstalling dependencies caused the same issue, however I fail to see how this PR affected that

**Comment by radimvaculik** [comment]
Is `compatibility.php` included in phpcs/phpstan paths somehow?

**Comment by lilianalillyy** [comment]
it's included in the composer autoload, however I doubt it has any impact on this as the error is regarding a Contributte namespace, and looking closely I see the issue, just not sure how it passed the CI before

**Comment by lilianalillyy** [comment]
It shouldn't work in the first place, as the signature of the parent class is in fact different

**Comment by radimvaculik** [comment]
But if you clone original repo, run `make` and `make cs`, it works like a charm 🤷‍♂️

**Comment by lilianalillyy** [comment]
i believe it's some sort of a bug: repro:
cs doesn't work on my next branch -> checkout to d7bbdacb738565bf0adef37059b179c33104fa0f (last commit of contributte's next branch -> make cs works -> checkout to e7b9b5e9907489512b6f35747a5208293d5b8b6e (last commit of my next branch) -> make cs works now

**Comment by lilianalillyy** [comment]
I have no idea what causes this, if someone could take a look it'd be great?

**Comment by lilianalillyy** [comment]
It's already shared in a gist above ^

**Comment by radimvaculik** [comment]
@lilianalillyy Let's go with for now.

```
-       public function getControl(string|object|null $caption = null): Html
+       /** @inheritDoc */
+       public function getControl($caption = null): Html
```

**Comment by lilianalillyy** [comment]
sounds good, let me update the code

File: src/Localization/SimpleTranslator.php [hunk 1]
```diff
@@ -1,44 +1,59 @@
...
 	{
+		// BC support for 'ublaboo' translations
+		$oldPrefix = 'ublaboo_';
+		$newPrefix = 'contributte_';
+		foreach ($dictionary as $key => $value) {
+			if (str_starts_with($key, $oldPrefix)) {
+				$newKey = $newPrefix . substr($key, strlen($oldPrefix));
+				// Only change the keys that are in the default $dictionary (other keys are left as-is)
+				if (array_key_exists($newKey, $this->dictionary)) {
+					trigger_error(`Translation key '$key' is deprecated, please use '$newKey' instead`, E_USER_DEPRECATED);
```
Comment (radimvaculik) [inline src/Localization/SimpleTranslator.php:50]: @lilianalillyy Use `sprintf`

**Review by radimvaculik — CHANGES_REQUESTED** [review]

**Comment by radimvaculik** [comment]
@f3l1x @paveljanda All good to me. Could you check PR and eventually merge it? Thanks guys.

**Review by radimvaculik — APPROVED** [review]

**Comment by radimvaculik** [comment]
@lilianalillyy Please, rebase 🙈

**Comment by lilianalillyy** [comment]
Will get to it later today!

**Comment by lilianalillyy** [comment]
@radimvaculik rebased! sorry it took so long, i was out of town (Brno :D)

**Comment by radimvaculik** [comment]
@lilianalillyy Thanks for contributing 👍

**Pull request merged**